### PR TITLE
Fixes in autotests C16994, C691 and C700

### PR DESCRIPTION
**Pull request by Sherzod-Kenjaev**
https://foliotest.testrail.io/index.php?/cases/view/16994
https://foliotest.testrail.io/index.php?/cases/view/691
https://foliotest.testrail.io/index.php?/cases/view/700

### CODE REVIEW

File: cypress/e2e/eholdings/eholdings-titles.cy.js [hunk 1]
```diff
@@ -47,9 +50,11 @@ describe('eHoldings titles management', () => {
     });
   });
 
-  // TODO: https://issues.folio.org/browse/UIEH-1256
+  //TODO: https://issues.folio.org/browse/UIEH-1256
```
Comment (MaksimTerekhov) [inline cypress/e2e/eholdings/eholdings-titles.cy.js:53]: Expected exception block, space or tab after '//' in comment

File: cypress/e2e/eholdings/eholdings-titles.cy.js [hunk 2]
```diff
@@ -21,8 +21,11 @@ describe('eHoldings titles management', () => {
...
-    cy.createTempUser([permissions.uieHoldingsRecordsEdit.gui,
-      permissions.uieHoldingsPackageTitleSelectUnselect.gui]).then(userProperties => {
+    cy.createTempUser([
+      permissions.uieHoldingsRecordsEdit.gui,
+      permissions.uieHoldingsPackageTitleSelectUnselect.gui,
+      permissions.uieHoldingsTitlesPackagesCreateDelete.gui
+    ]).then(userProperties => {
       userId = userProperties.userId;
       cy.login(userProperties.username, userProperties.password);
       cy.visit(TopMenu.eholdingsPath);
```
Comment (MaksimTerekhov) [inline cypress/e2e/eholdings/eholdings-titles.cy.js:31]: cy.login(userProperties.username, userProperties.password, { path: ..., waiter: ... });

File: cypress/e2e/eholdings/eholdings-titles.cy.js [hunk 3]
```diff
@@ -47,9 +50,11 @@ describe('eHoldings titles management', () => {
...
-  // TODO: https://issues.folio.org/browse/UIEH-1256
+  //TODO: https://issues.folio.org/browse/UIEH-1256
   it('C700 Title: Add or Edit custom coverage (spitfire)', { tags: [testTypes.smoke, devTeams.spitfire, features.eHoldings] }, () => {
-    cy.createTempUser([permissions.uieHoldingsRecordsEdit.gui]).then(userProperties => {
+    cy.createTempUser([
+      permissions.uieHoldingsRecordsEdit.gui
+    ]).then(userProperties => {
       userId = userProperties.userId;
       cy.login(userProperties.username, userProperties.password);
       cy.visit(TopMenu.eholdingsPath);
```
Comment (MaksimTerekhov) [inline cypress/e2e/eholdings/eholdings-titles.cy.js:60]: cy.login(userProperties.username, userProperties.password, { path: ..., waiter: ... });

File: cypress/support/fragments/eholdings/eHoldingsTitle.js [hunk 4]
```diff
@@ -16,27 +18,29 @@ export default {
   filterStatuses,
 
   filterPackages: (selectionStatus = filterStatuses.notSelected, packageName) => {
+    //Test is not working stabil if no waits.
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/eholdings/eHoldingsTitle.js:21]: Here we need a comment why you use cy.wait here and why you can not use element expectation with interactor

File: cypress/e2e/eholdings/eholdings-titles.cy.js (hunk 1 above)
Comment (Sherzod-Kenjaev) [reply cypress/e2e/eholdings/eholdings-titles.cy.js]: fixed

File: cypress/e2e/eholdings/eholdings-titles.cy.js (hunk 2 above)
Comment (Sherzod-Kenjaev) [reply cypress/e2e/eholdings/eholdings-titles.cy.js]: fixed

File: cypress/e2e/eholdings/eholdings-titles.cy.js (hunk 3 above)
Comment (Sherzod-Kenjaev) [reply cypress/e2e/eholdings/eholdings-titles.cy.js]: fixed

File: cypress/support/fragments/eholdings/eHoldingsTitle.js (hunk 4 above)
Comment (Sherzod-Kenjaev) [reply cypress/support/fragments/eholdings/eHoldingsTitle.js]: Fixed. Test is not running stable with element expectation, there is a method before this one to wait until element  is found and it is finding and clicking but the modal is not loading sometimes. When I use wait 2 seconds it is working properly.

File: cypress/e2e/eholdings/eholdings-titles.cy.js [hunk 5]
```diff
@@ -21,11 +21,13 @@ describe('eHoldings titles management', () => {
...
-      permissions.uieHoldingsPackageTitleSelectUnselect.gui]).then(userProperties => {
+    cy.createTempUser([
+      permissions.uieHoldingsRecordsEdit.gui,
+      permissions.uieHoldingsPackageTitleSelectUnselect.gui,
+      permissions.uieHoldingsTitlesPackagesCreateDelete.gui
+    ]).then(userProperties => {
       userId = userProperties.userId;
-      cy.login(userProperties.username, userProperties.password);
-      cy.visit(TopMenu.eholdingsPath);
+      cy.login(userProperties.username, userProperties.password, { path: TopMenu.eholdingsPath, waiter: () => eHoldingsTitlesSearch.waitLoading() });
```
Comment (MaksimTerekhov) [inline cypress/e2e/eholdings/eholdings-titles.cy.js:30]: you can write just

`waiter: eHoldingsTitlesSearch.waitLoading`

there is no need arrow function

File: cypress/e2e/eholdings/eholdings-titles.cy.js [hunk 6]
```diff
@@ -49,10 +51,11 @@ describe('eHoldings titles management', () => {
...
   // TODO: https://issues.folio.org/browse/UIEH-1256
   it('C700 Title: Add or Edit custom coverage (spitfire)', { tags: [testTypes.smoke, devTeams.spitfire, features.eHoldings] }, () => {
-    cy.createTempUser([permissions.uieHoldingsRecordsEdit.gui]).then(userProperties => {
+    cy.createTempUser([
+      permissions.uieHoldingsRecordsEdit.gui
+    ]).then(userProperties => {
       userId = userProperties.userId;
-      cy.login(userProperties.username, userProperties.password);
-      cy.visit(TopMenu.eholdingsPath);
+      cy.login(userProperties.username, userProperties.password, { path: TopMenu.eholdingsPath, waiter: () => eHoldingsTitlesSearch.waitLoading() });
```
Comment (MaksimTerekhov) [inline cypress/e2e/eholdings/eholdings-titles.cy.js:58]: the same about waiter

File: cypress/e2e/eholdings/eholdings-titles.cy.js [hunk 7]
```diff
@@ -113,12 +119,13 @@ describe('eHoldings titles management', () => {
...
-      permissions.uieHoldingsTitlesPackagesCreateDelete.gui]).then(userProperties => {
+    cy.createTempUser([
+      permissions.uieHoldingsRecordsEdit.gui,
+      permissions.uieHoldingsTitlesPackagesCreateDelete.gui
+    ]).then(userProperties => {
       userId = userProperties.userId;
       eHoldingsPackages.getCustomPackageViaApi().then(packageName => {
-        cy.login(userProperties.username, userProperties.password);
-        cy.visit(TopMenu.eholdingsPath);
+        cy.login(userProperties.username, userProperties.password, { path: TopMenu.eholdingsPath, waiter: () => eHoldingsTitlesSearch.waitLoading() });
```
Comment (MaksimTerekhov) [inline cypress/e2e/eholdings/eholdings-titles.cy.js:128]: the same about waiter

**Review by MaksimTerekhov — APPROVED** [review]

**Review by OllIsanbaev — APPROVED** [review]

**Pull request merged**